modules/linear_boltzmann_solvers/lbs_solver/lbs_solver.cc

```diff
@@ -797,17 +789,17 @@ LBSSolver::Initialize()
 {
   CALI_CXX_MARK_SCOPE("LBSSolver::Initialize");
 
-  PerformInputChecks(); // a assigns num_groups and grid
-  PrintSimHeader();     // b
+  PerformInputChecks(); // assigns num_groups and grid
```

Comment:
If `PerformInputChecks` really assigns num_groups and grid then it is either a) misnomer or b) bad design.

Reply:
I think it 's a bit of both. It looks like it does initialization and checks for various quantities. I'd like to invest more time into refactoring this entire class, so I think I'll defer this issue until after 1.0.

Reply:
Ok, no problem :+1: